### PR TITLE
fix: add nuxt generate support

### LINKED ISSUE #7: Service workers are not generated with static build
Currently, when I do a `nuxt generate`, the service worker is generated (sw.js and workbox file) in .output/public but remove after that during the generation process.

## Configuration

```ts

export default defineNuxtConfig({
  modules: [
    '@vite-pwa/nuxt'
  ],
  pwa: {
    manifest: false,
    injectRegister: 'auto'
  }
})
```

### PR DESCRIPTION
**Pull request by userquin**
This PR also includes: throw error when using webpack

closes #7

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -121,11 +125,17 @@ export default defineNuxtModule<ModuleOptions>({
...
       nuxt.hook('nitro:init', (nitro) => {
         nitro.hooks.hook('rollup:before', async () => {
+          swGenerated = true
           await resolveVitePluginPWAAPI()?.generateSW()
         })
       })
+      nuxt.hook('close', async () => {
+        if (!swGenerated)
+          await resolveVitePluginPWAAPI()?.generateSW()
+      })
```
Comment (danielroe) [inline src/module.ts:138]: Was this not being generated before?

Comment (userquin) [reply src/module.ts:138]: No

Comment (userquin) [reply src/module.ts:138]: When running `nuxt generate`, `rollup:before` hook not being called and so only the client build is generated, missing all html pages from the sw precache manifest.

Comment (danielroe) [inline src/module.ts:138]: We could also do it like this:

```suggestion
      if (nuxt.options._generate) {
        nuxt.hook('close', async () => {
          await resolveVitePluginPWAAPI()?.generateSW()
        })
      }
```

**Review by danielroe — APPROVED** [review]

**Pull request merged**